contracts/Vesting.sol

```diff
@@ -0,0 +1,51 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.6.0;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts-ethereum-package/contracts/math/SafeMath.sol";
+
+contract Vesting is Ownable {
+
+    using SafeMath for uint;
+
+    uint constant NUMBER_OF_EPOCHS = 100;
+    uint constant EPOCH_DURATION = 604800; // 1 week duration
+    IERC20 _bond;
+
+    uint public lastClaimedEpoch;
+    uint private _startTime;
+    uint public totalDistributedBalance;
+
+    constructor(address newOwner, address bondTokenAddress, uint startTime, uint totalBalance) public {
+        transferOwnership(newOwner);
+        _bond = IERC20(bondTokenAddress);
+        _startTime = startTime;
+        totalDistributedBalance = totalBalance;
+    }
+
+    function claim () public onlyOwner {
+        uint balance;
+        uint currentEpoch = getCurrentEpoch();
+        if (currentEpoch > NUMBER_OF_EPOCHS + 1) {
+            currentEpoch = NUMBER_OF_EPOCHS + 1;
```

Comment:
I think here we should just transfer all the balance of the contract to owner (if > 0), in case there is any dust left.